warehouse/templates/manage/history.html

```diff
@@ -17,38 +17,64 @@
+  <p>Each time you (or your collaborators) perform a security action related to this project, the action is recorded and displayed here.</p>
 
-  <table class="table table--history margin-top">
-    <caption class="sr-only">History for {{ project.name }}</caption>
+  {% macro event_summary(event) -%}
+    {% if event.tag == "project:api_token:added" %}
+    <strong>API token added</strong><br>
+    <small>
+      Permissions: Can upload to this project<br>
+      Controlled by: <a href="{{ request.route_path('accounts.profile', username=event.additional.user) }}">{{ event.additional.user }}</a><br>
+      Token name: {{ event.additional.description }}
+    </small>
+    {% elif event.tag == "project:api_token:removed" %}
+    <strong>API token removed</strong><br>
+    <small>
+      Permissions: Can upload to this project<br>
+      Controlled by: <a href="{{ request.route_path('accounts.profile', username=event.additional.user) }}">{{ event.additional.user }}</a><br>
+      Token name: {{ event.additional.description }}
+    </small>
+    {% elif event.tag == "project:role:add" %}
+    <strong><a href="{{ request.route_path('accounts.profile', username=event.additional.target_user) }}">{{ event.additional.target_user }}</a> added as project {{ event.additional.role_name|lower }}</strong><br>
+    <small>
+      Added by: <a href="{{ request.route_path('accounts.profile', username=event.additional.submitted_by) }}">{{ event.additional.submitted_by }}</a>
+    </small>
+    {% elif event.tag == "project:role:delete" %}
+    <strong><a href="{{ request.route_path('accounts.profile', username=event.additional.target_user) }}">{{ event.additional.target_user }}</a> removed as project {{ event.additional.role_name|lower }}</strong><br>
+    <small>
+      Removed by: <a href="{{ request.route_path('accounts.profile', username=event.additional.submitted_by) }}">{{ event.additional.submitted_by }}</a>
+    </small>
+    {% elif event.tag == "project:role:change" %}
+    <strong><a href="{{ request.route_path('accounts.profile', username=event.additional.target_user) }}">{{ event.additional.target_user }}</a> changed to project {{ event.additional.role_name|lower }}</strong><br>
+    <small>
+      Changed by: <a href="{{ request.route_path('accounts.profile', username=event.additional.submitted_by) }}">{{ event.additional.submitted_by }}</a>
+    </small>
+    {% endif %}
+  {%- endmacro %}
+
+  <table class="table table--light table--security-logs">
+    <caption class="sr-only">Security history for {{ project.name }}</caption>
     <thead>
       <tr>
-        <th scope="col">Action</th>
-        <th scope="col">Date</th>
-        <th scope="col">User</th>
+        <th scope="col">Event</th>
+        <th scope="col">Date / time</th>
+        <th scope="col">IP address</th>
       </tr>
     </thead>
     <tbody>
-      {% for journal in journals %}
+      {% for event in events %}
```

Comment:
I think we want to keep Journals visible, and display them in a separate table. Once more, to expose this information until we either migrate JournalEntry objects into ProjectEvent objects or otherwise deprecate them.

Reply:
Thanks @ewdurbin.  FYI we have a ticket discussing this here : https://github.com/pypa/warehouse/issues/6425

Reply:
Just found that and commented. 😬